src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -5,3 +5,265 @@ title: Build System
+
+```sh
+./b format
+```
+
+##### Install
+
+Once code is built and you are ready to install it on a robot, you can run
+
+```sh
+./b install [options] <robot>
+```
+
+`<robot>` can either be a known robot designation (`n1` or `nugus1`, for instance) or an IP address of a robot.
+
+`[options]` may be one, or more, of the following
+
+| Option | Description                                                                    |
+| :----: | :----------------------------------------------------------------------------- |
+|   -u   | The user to install to on the target. Defaults to the user in the Docker image |
+|   -t   | Install toolchain to the target                                                |
+|  -cn   | Only install new config files. This is the default                             |
+|  -cu   | Update config files on the target that are older than the local files          |
+|  -co   | Overwrite all config files on the target                                       |
+|  -ci   | Ignore all changes to config files (installs no config files)                  |
+
+##### Module
+
+The `module` command allows you to generate a new module. To generate a new line detection module one would run
+
+```sh
+./b module generate vision/LineDetector
+```
+
+This will generate all necessary module files in the `module/vision/LineDetector` folder. This includes an empty configuration file, a basic C++ source and header file implementing the `LineDetector` reactor, and an empty unit test file.
+
+##### Run
+
+To run roles locally on your system use the `run` command of `b` as follows
+
+```sh
+./b run <name of role>
+```
+
+---
+
+**NOTE**
+Be sure that you are using the `generic` target for this (`./b target generic`), otherwise you role will likely crash.
+
+---
+
+##### Target
+
+The `target` command of `b` allows you to build code for a different target. Currently, only two targets are supported: `generic` and `nuc7i7bnh`.
+
+The `generic` target is useful if you would like to build and run code on your local machine.
+
+The `nuc7i7bnh` target should be used for building and running code on the robots.
+
+The `target` command will download (or, if needed, build) the Docker image for the specified target and then mark that target as active.
+
+## Continuous Integration
+
+NUbots uses [Buildkite](buildkite.com) for its continuous integration pipeline. The Buildkite server will issue build jobs to one of our build clients (the Macs in the lab) as new jobs become available.
+
+The build flow that the NUbots uses a pull request system. Developers branch off of `master`, creating a new branch named `<last name>/<purpose of branch>`. Once development on the branch is complete, the developer makes a pull request and a job is issued to a Buildkite client. Each pull request must be reviewed by another member of the team and all build checks on the Buildkite client must pass before the pull request can be approved and merged into `master`.
```

Comment:
Is this really a part of the `build system` (I could be convinced either way)
Also, in general, I'm not sure if it's done in this repo but git diffs become much easier if you have a sentence per line

Reply:
It was already in the title when I started, if you think it should be moved then we can make another page for it

Reply:
Might be a good idea, I'm not really sure the two of them are that related from a users perspective.